serf/snapshot_test.go

```diff
@@ -388,3 +389,145 @@ func TestSnapshotter_leave_rejoin(t *testing.T) {
 		t.Fatalf("expected alive: %#v", prev)
 	}
 }
+
+func TestSnapshotter_slowDiskNotBlockingEventCh(t *testing.T) {
+	td, err := ioutil.TempDir("", "serf")
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	t.Log("Temp dir", td)
+	defer os.RemoveAll(td)
+
+	clock := new(LamportClock)
+	stopCh := make(chan struct{})
+	logger := log.New(os.Stderr, "", log.LstdFlags)
+
+	outCh := make(chan Event, 1024)
+	inCh, snap, err := NewSnapshotter(td+"snap", snapshotSizeLimit, true,
+		logger, clock, outCh, stopCh)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// We need enough events to be much more than the buffers used which are size
+	// 1024. This number processes easily within the 500ms we allow below on my
+	// host provided there is no disk IO on the path (I verified that by just
+	// returning early in tryAppend using the old blocking code). The new async
+	// method should pass without disabling disk writes too!
+	numEvents := 10000
+
+	// Write lots of member updates (way bigger than our chan buffers)
+	startCh := make(chan struct{})
+	go func() {
+		<-startCh
+		for i := 0; i < numEvents; i++ {
+			e := MemberEvent{
+				Type: EventMemberJoin,
+				Members: []Member{
+					Member{
+						Name: fmt.Sprintf("foo%d", i),
+						Addr: []byte{127, 0, byte((i / 256) % 256), byte(i % 256)},
+						Port: 5000,
+					},
+				},
+			}
+			if i%10 == 0 {
+				// send a leave for every 10 joins
```

Comment:
:nit

Technically the comment isn't accurate. For every 9 joins this will send a leave.